internal/store/hpa.go

```diff
@@ -76,6 +78,73 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_spec_target_metric",
+			Type: metric.Gauge,
+			Help: "The metric specifications used by this autoscaler when calculating the desired replica count.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				targets := make([]*metric.Metric, 0, len(a.Spec.Metrics))
+				for _, m := range a.Spec.Metrics {
+					var metricName string
+					var value, average int64
+					valueLabel := "value"
+					var valueOk, averageOk bool
+
+					switch m.Type {
+					case autoscaling.ObjectMetricSourceType:
+						metricName = m.Object.MetricName
+
+						value, valueOk = m.Object.TargetValue.AsInt64()
+						if m.Object.AverageValue != nil {
+							average, averageOk = m.Object.AverageValue.AsInt64()
+						}
+					case autoscaling.PodsMetricSourceType:
+						metricName = m.Pods.MetricName
+
+						average, averageOk = m.Pods.TargetAverageValue.AsInt64()
+					case autoscaling.ResourceMetricSourceType:
+						metricName = string(m.Resource.Name)
+
+						value = int64(*m.Resource.TargetAverageUtilization)
+						valueOk = true
+						valueLabel = "utilization"
```

Comment:
I use the valueLabel var here, should I drop the "utilization" label value?

Reply:
Hmmmmm, I think having "utilization" as the metric value is important. It would be misleading otherwise.
You can probably do either of the two
i) Have a new pair of vars `utilization` and `utilizationOk` (could shorten it to `utilz` and `utilzOk`.

OR

ii) Have an custom enum Type to represent the three types of metric vals - `value`, `average`, and `utilization`.

Your call :).

Reply:
Went the enum route 😄

Reply:
Awesome!